tests/test_m2m.py

```diff
@@ -0,0 +1,278 @@
+import unittest
+
+from lxml import etree
+from lxml import objectify
+
+from m2m import m2m
+
+
+def xml_to_pretty_string(xml):
+    objectify.deannotate(xml, xsi_nil=True)
+    etree.cleanup_namespaces(xml)
+
+    s = etree.tostring(xml, pretty_print=True)
+    return '<?xml version="1.0" encoding="UTF-8"?>\n' + s
+
+
+class MetadataRecordTests(unittest.TestCase):
+
+    def setUp(self):
+        xml = """<?xml version="1.0" encoding="UTF-8"?>
+            <metadata>
+                <title />
+                <meta qualifier='metadataCreator'></meta>
+            </metadata>
+        """
+        self.tree = objectify.fromstring(xml)
+
+        agent_xml = """<?xml version="1.0" encoding="UTF-8"?>
+            <metadata>
+                <creator>
+                    <info />
+                    <type />
+                    <name />
+                </creator>
+                <meta />
+            </metadata>
+        """
+        self.agent_tree = objectify.fromstring(agent_xml)
+
+        pub_xml = """<?xml version="1.0" encoding="UTF-8"?>
+            <metadata>
+                <publisher>
+                    <name />
+                    <location />
+                </publisher>
+                <meta />
+            </metadata>
+        """
+        self.pub_tree = objectify.fromstring(pub_xml)
+
+    def test_metadata_record_setup(self):
+        # test that the creation of a record creates a MetadataRecord object.
+
+        record = m2m.MetadataRecord('mphillips')
+        self.assertIsInstance(record, m2m.MetadataRecord)
+
+    def test_metadata_record_setup_with_date(self):
+
+        record = m2m.MetadataRecord('mphillips', addDate=True)
+        self.assertIsInstance(record, m2m.MetadataRecord)
```

Comment:
It might be useful to test if there is now a meta tag with qualifier "metadataCreationDate" in the record instead of just that this is a record.